r/tests/testthat/test-dataset.R

```diff
@@ -453,15 +453,38 @@ test_that("Creating UnionDataset", {
 })
 
 test_that("map_batches", {
-  skip("map_batches() is broken (ARROW-14029)")
   ds <- open_dataset(dataset_dir, partitioning = "part")
+
+  # summarize returns arrow_dplyr_query
   expect_equal(
     ds %>%
       filter(int > 5) %>%
       select(int, lgl) %>%
-      map_batches(~ summarize(., min_int = min(int))),
+      map_batches(~ summarize(., min_int = min(int))) %>%
+      arrange(min_int),
     tibble(min_int = c(6L, 101L))
   )
+
+  # $num_rows returns integer vector
+  expect_equal(
+    ds %>%
+      filter(int > 5) %>%
+      select(int, lgl) %>%
+      map_batches(~ .$num_rows, .data.frame = FALSE) %>%
+      as.numeric() %>%
+      sort(),
+    c(5, 10)
+  )
+
+  # $Take returns RecordBatch
```

Comment:
Is this comment accurate here? It looks like it's returning a tibble? Or is that a side effect of `arrange()`?

Reply:
It returns a record batch within that function. But since the `.data.frame` option on `map_batches()` is `TRUE` by default, the results are combined into a tibble using `dplyr::bind_rows()`.

Reply:
Ah I see, it might be good to add that to the comment so that it's clear what's going on there.

Reply:
Yes, I've added a couple clarifying comments.